packages/pyright-internal/src/analyzer/importResolver.ts

```diff
@@ -1120,9 +1121,13 @@ export class ImportResolver {
             importFailureInfo.push('No python interpreter search path');
         }
 
-        if (bestResultSoFar?.pyTypedInfo && !bestResultSoFar.isPartlyResolved) {
-            // If a library is fully py.typed, then we have found the best match.
-            return bestResultSoFar;
+        // Special case: If the execution environment is typeshed itself, don't
+        // try and use a py.typed library from the search paths.
```

Comment:
I find this comment a bit confusing because we will still use a py.typed library from the search paths if we don't find a better match. I think what you mean to say is that we won't favor a py.typed library over a typeshed stub in this circumstance.

Reply:
Updated the comment, if you want to look again.

Reply:
Ha, actually pushed now.